chain-test/src/test/scala/org/bitcoins/chain/util/ChainUnitTest.scala

```diff
@@ -94,4 +99,59 @@ trait ChainUnitTest
 
     dropTableP.future.flatMap(_ => testExecutionF)
   }
+
+  /** Represents a bitcoind instance paired with a chain handler via zmq */
+  case class BitcoindChainHandler(
+      bitcoindRpc: BitcoindRpcClient,
+      chainHandler: ChainHandler)
+
+  def bitcoindZmqChainHandler(test: BitcoindChainHandler => Future[Assertion])(
+      implicit system: ActorSystem): Future[Assertion] = {
+    val instance = BitcoindRpcTestUtil.instance()
+    val bitcoindF = {
+      val bitcoind = new BitcoindRpcClient(instance)
+      bitcoind.start().map(_ => bitcoind)
+    }
+    val zmqRawBlockUriF = bitcoindF.map(_ => instance.zmqConfig.rawBlock)
+    val f: ChainHandler => Future[Assertion] = { chainHandler: ChainHandler =>
+      val handleRawBlock: ByteVector => Unit = { bytes: ByteVector =>
+        val block = Block.fromBytes(bytes)
+        chainHandler.processHeader(block.blockHeader)
+
+        ()
+      }
+
+      val zmqSubscriberF = zmqRawBlockUriF.map { uriOpt =>
+        val socket = uriOpt.get
```

Comment:
Looking at what config `BitcoindRpcTestUtil.instance()` uses this is safe, but kinda smelly that we have to do a `.get` here. Not necessarily in the scope of this PR, though